macros/contexts/contextNondecimalBase.pl

```diff
@@ -0,0 +1,268 @@
+
+=head1 NAME
+
+contextNonDecimalBase.pl - Implements a MathObject class and context for numbers
+in non-decimal bases
+
+=head1 DESCRIPTION
+
+This context implements positive integers and some operations on integers in a non-decimal base
+greater than or equal to 2.  The numbers will be stored internally in decimal, though parsed
+and shown in the chosen base.
+
+In addition, basic integer arithemetic (+,-,*,^) are available for these numbers, however because it
+is assumed that all numbers are integers, division '/' is disallowed.
+
+The original purpose for this is simple conversion and operations in another base, however
+it is not limited to this.
+
+To use a non-decimal base MathObject, first load the contextNondecimalBase.pl file:
+
+    loadMacros('contextNondecimalBase.pl');
+
+and then select the appropriate context -- one of the following:
+
+    Context('NondecimalBase')->setBase(5);
+
+
+will now interprets those in Compute and student answers in base 5.
+
+    $a = Compute('104');
+    $b = Compute('233');
+    $sum = Compute("$a+$b"); # this is the base-5 number 342 (decimal 97)
```

Comment:
There is no need to use `Compute()`, here, as you can just do `$sum = $a + $b` and avoid the multiple conversions to/from strings.  You could use

``` perl
$sum = Compute("104 + 233");
```

as a better example.

Reply:
Makes sense!